fileglancer_central/database.py

```diff
@@ -158,15 +161,50 @@ def initialize_database(db_url):
     run_alembic_upgrade(db_url)
 
 
+def _get_engine(db_url):
+    """Get or create a cached database engine for the given URL"""
+    global _engine_cache
+    
+    # Return cached engine if it exists
+    if db_url in _engine_cache:
+        return _engine_cache[db_url]
+    
+    # Create new engine and cache it
+    engine = create_engine(
+        db_url,
+        pool_size=10,
+        max_overflow=20,
+        pool_recycle=3600,  # Recycle connections after 1 hour
+        pool_pre_ping=True  # Verify connections before use
+    )
```

Comment:
Does this connection pool work for Sqlite? I don't think it will because Sqlite is single threaded. We may need to check if Sqlite is being used and use an alternative engine here.